Shopify/dawn
# [Slideshow] Add fixed position background

### PR Summary: 

Adds a fixed position background option to the Slideshow section. 

https://user-images.githubusercontent.com/1202812/221970091-6166b1e5-e574-4623-910a-49595553bcd2.mp4

### Why are these changes introduced?

Closes #2345.

### What approach did you take?

This implements a new dropdown menu [as shown in the ticket](https://github.com/Shopify/dawn/issues/2345). When "Fixed background position" is selected, a CSS class is applied to the media elements within the Slideshow slides. This class sets the position of the media element to `fixed`, and uses `clip-path(0)` on the parent to ensure the image doesn't bleed outside of its intended container. 

The CSS for this is defined in `base.css` so that we can theoretically reuse it for the Image with Text (https://github.com/Shopify/dawn/issues/2340) and Image Banner (https://github.com/Shopify/dawn/issues/2341) sections.

### Other considerations

 ⚠ We may want to re-evaluate this approach, since it's causing some issues in Chrome. Sometimes, the pseudo element and/or slideshow content doesn't appear correctly when you scroll into it or advance onto the next slide. Scrolling a little or moving the mouse near the slideshow fixes this. From what I can infer, this is related to the use of `clip-path`. These issues are not present in Safari or Firefox. 😞 

If for some reason this approach falls through, we could also explore setting the `background-image` of the media container, and using `background-attachment: fixed`. This should avoid the issues, but is far more verbose and likely less performant, as it eliminates the ability for us to use `srcset`. 😕

<details>
<summary>📹 Example</summary>

https://user-images.githubusercontent.com/1202812/221972454-13d971eb-eb8c-4f38-b487-05ec369184f3.mp4

</details>


### Visual impact on existing themes

None. This is an optional feature. 

### Testing steps/scenarios

1. [Visit the demo store](https://os2-demo.myshopify.com/admin/themes/139452514326/editor).
2. Add and select a Slideshow section.
3. Look for the new "Image behavior" dropdown underneath the new "Animations" header in the sidebar.
4. Select "Fixed position background" from that dropdown.
5. Add various images, verify that they appear as expected.
6. Try various breakpoints, verify that they appear as expected.

### Checklist
